src/components/store/AnalyzeStore.ts

```diff
@@ -16,44 +17,75 @@ interface AnalyzeActions {
 	}
 }
 
-// 카테고리별 뭉치고 가격 합산
-const combiningData = (
-	consumptionData: analyzeDataType[],
-): analyzeCombinedDataType[] => {
-	let id = 0
-	const consumptionCombinedData = Object.values(
-		consumptionData.reduce<Record<string, analyzeCombinedDataType>>(
-			(acc, item) => {
-				if (!acc[item.category]) {
-					acc[item.category] = {
-						id: id++,
-						category: item.category,
-						price: item.price,
-					}
-				} else {
-					acc[item.category].price += item.price
-				}
-				return acc
+function createDataItems(data: {
+	[key: string]: { value: number; firstId: number }
+}): spendingDataProps[] {
+	return Object.entries(data)
+		.sort(([, a], [, b]) => b.value - a.value)
+		.map(([key, { value, firstId }], index) => ({
+			id: firstId,
+			data: {
+				id: key,
+				label: key,
+				value,
+				color: chartColors[index % chartColors.length],
 			},
-			{},
-		),
-	)
-	return consumptionCombinedData.sort((a, b) => b.price - a.price)
+		}))
 }
 
+function processConsumptionData(
+	data: any[],
```

Comment:
`any` 는 타입스크립트의 이점인 타입 안정성을 잃게 되는 것 같습니다.
받는 데이터를 보니
```ts
const consumptionMockData: {
    id: number;
    input_type: string;
    amount: number;
    place: string;
    memo: string;
    date: string;
    category: {
        id: number;
        value: string;
        name: string;
    };
}[]
```
이런 형식으로 들어오는 것 같은데, `IInputFormData` 의 타입과 유사하지만 store 내에서 사용하실 거니까 따로 타입을 두어 정해주거나 extends 해서 정의해도 괜찮을 것 같아요.
```suggestion
const ISpendingItem {
    id: number;
    input_type: string;
    amount: number;
    place: string;
    memo: string;
    date: string;
    category: ICategory
 }
 ...
	data: ISpendingItem[],
```

Reply:
수정하려고 했는데 까먹었네요 조심하겠습니다.